cli/internal/cmd/verify.go

```diff
@@ -186,24 +186,28 @@ func (c *verifyCmd) parseVerifyFlags(cmd *cobra.Command, fileHandler file.Handle
 	}
 	c.log.Debugf("Flag 'raw' set to %t", force)
 
-	var idFile clusterid.File
-	if err := fileHandler.ReadJSON(constants.ClusterIDsFilename, &idFile); err != nil && !errors.Is(err, afero.ErrFileNotFound) {
-		return verifyFlags{}, fmt.Errorf("reading cluster ID file: %w", err)
-	}
-
-	// Get empty values from ID file
+	// Get empty values from state file
+	var attestationURL string
 	emptyEndpoint := endpoint == ""
 	emptyIDs := ownerID == "" && clusterID == ""
 	if emptyEndpoint || emptyIDs {
-		c.log.Debugf("Trying to supplement empty flag values from %q", pf.PrefixPrintablePath(constants.ClusterIDsFilename))
+		c.log.Debugf("Trying to supplement empty flag values from %q", pf.PrefixPrintablePath(constants.StateFilename))
+		stateFile, err := state.ReadFromFile(fileHandler, constants.StateFilename)
+		if err != nil {
+			return verifyFlags{}, fmt.Errorf("reading state file: %w", err)
+		}
+
 		if emptyEndpoint {
-			cmd.Printf("Using endpoint from %q. Specify --node-endpoint to override this.\n", pf.PrefixPrintablePath(constants.ClusterIDsFilename))
-			endpoint = idFile.IP
+			cmd.Printf("Using endpoint from %q. Specify --node-endpoint to override this.\n", pf.PrefixPrintablePath(constants.StateFilename))
+			endpoint = stateFile.Infrastructure.ClusterEndpoint
 		}
 		if emptyIDs {
-			cmd.Printf("Using ID from %q. Specify --cluster-id to override this.\n", pf.PrefixPrintablePath(constants.ClusterIDsFilename))
-			ownerID = idFile.OwnerID
-			clusterID = idFile.ClusterID
+			cmd.Printf("Using ID from %q. Specify --cluster-id to override this.\n", pf.PrefixPrintablePath(constants.StateFilename))
+			ownerID = stateFile.ClusterValues.OwnerID
+			clusterID = stateFile.ClusterValues.ClusterID
+		}
+		if stateFile.Infrastructure.Azure != nil {
+			attestationURL = stateFile.Infrastructure.Azure.AttestationURL
```

Comment:
Performing this check inside the `if emptyEndpoint || emptyIDs` block, will cause `attestationURL` to always be an empty string if a user specifies IDs and endpoint manually

Reply:
Since we don't want a failure if the arguments are specified but no state-file is present, how about something like this?

```go
var attestationURL string
	stateFile, err := state.ReadFromFile(fileHandler, constants.StateFilename)
	if err == nil {
		if stateFile.Infrastructure.Azure != nil {
			attestationURL = stateFile.Infrastructure.Azure.AttestationURL
		}
	}
```

`if err == nil``` smells like an antipattern though

Reply:
You could just keep the same logic as before, but replace ID file with state file.